frontend/src/component/admin/network/NetworkTraffic/NetworkTraffic.tsx

```diff
@@ -123,65 +117,66 @@ const createInstanceChartOptions = (
         },
     };
 };
+class ItemPicker<T> {
+    private items: CyclicIterator<T>;
+    private picked: Map<string, T> = new Map();
+    constructor(items: T[]) {
+        this.items = new CyclicIterator<T>(items);
+    }
 
-const toChartData = (
-    rps: RequestsPerSecondSchema,
-    color: PaletteColor,
-    label: (name: string) => string
-): ChartDatasetType[] => {
-    if (rps.data?.result) {
-        return rps.data.result.map(dataset => ({
-            label: label(dataset.metric?.appName || 'unknown'),
-            borderColor: color.main,
-            backgroundColor: color.main,
-            data: createChartPoints(dataset.values || [], y => parseFloat(y)),
-            elements: {
-                point: {
-                    radius: 4,
-                    pointStyle: 'circle',
-                },
-                line: {
-                    borderDash: [8, 4],
-                },
-            },
-        }));
+    public pick(key: string): T {
+        if (!this.picked.has(key)) {
+            this.picked.set(key, this.items.next());
+        }
+        return this.picked.get(key)!;
     }
-    return [];
-};
+}
 
-const createInstanceChartData = (metrics?: InstanceMetrics): ChartDataType => {
-    if (metrics) {
-        const colors = new CyclicIterator<PaletteColor>([
+const toChartData = (rps?: RequestsPerSecondSchema): ChartDatasetType[] => {
+    if (rps?.data?.result) {
+        const colorPicker = new ItemPicker([
             theme.palette.success,
             theme.palette.error,
             theme.palette.primary,
+            theme.palette.warning,
         ]);
-        let datasets: ChartDatasetType[] = [];
-        for (let key in metrics) {
-            datasets = datasets.concat(
-                toChartData(
-                    metrics[key],
-                    colors.next(),
-                    metricName => `${metricName}: ${key}`
-                )
-            );
-        }
-        return { datasets };
+        return rps.data.result.map(dataset => {
+            const endpoint = dataset.metric?.endpoint || 'unknown';
+            const appName = dataset.metric?.appName || 'unknown';
```

Comment:
This can be 'undefined'  as well, we need to check if we want to have only 'unknown'

Reply:
Used: `(dataset.metric?.appName!).replace(/^undefined$/, 'unknown');`